File: taxi.py
import psycopg2
from tabulate import tabulate

def  post(x,y,z):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO taxi(modelo, recorrido_Km, estado) VALUES (%s, %s, %s);", (x, y, z))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from taxi;")
    print(tabulate(cursor, headers=["ide", "modelo", "recorrido_km", "estado"], tablefmt="orgtbl", numalign ="center"))

while True:
    try:
        print("Elija una opcion: (1. Saber la condicion del taxi,   2.Historial,   3.Salir)")
        x=int(input("Elija la opcion:\n"))
        if x>3 or x<=0:
            print("Valor ingresado fuera del rango del menu.\n")
        elif x==1:
            estado=""
            modelo=int(input("Ingrese el modelo del taxi: "))
            recorrido=int(input("Ingrese el recorrido en Km del taxi: "))
            if modelo<2007 and 20<recorrido:
                print("Debe renovarse")
                estado="Debe renovarse"
            
            elif 2007<=modelo<=2013 and 20000==recorrido:
                print("Debe recibir mantenimiento")
                estado="Debe recibir mantenimiento"

            elif modelo>2013 and 10000>recorrido:
                print("Esta en optimas condiciones")
                estado="Esta en optimas condiciones"
            else:
                print("Mecánico")
                estado="Mecánico"
            post(modelo, recorrido, estado)
        elif x==2:
            ret()
        elif x==3:
            break
    except Exception as e:
        print("\n A ingresado un valor o Caracter no valido por favor, ingrese una opcion correcta:\n")
        print(repr(e))

exit(0)

File: taxi.m
clc; 
clear all; 
close all;

retry= true;
while retry
  try
    fprintf("Elija una opcion: (1. Saber la condicion del taxi,   2.Historial)");
    i=(input ("\n Opcion a elegir: "));
    if i>3||i<0
      fprintf("\n Numero erroneo, vuelva a ingresar: ");
    endif
    if i==1
      modelo=(input('Ingrese el modelo del taxi: '));
      recorrido=(input('Ingrese el recorrido en Km del taxi: '));
      if modelo<2007&&20<recorrido
         fprintf('Debe renovarse \n');
         estado='Debe renovarse'; 
      elseif 2007<=modelo<=2013&&20000==recorrido
        fprintf('Debe recibir mantenimiento \n');
        estado='Debe recibir mantenimiento';
      elseif modelo>2013&&10000>recorrido
        fprintf('Esta en optimas condiciones \n');
        estado='Esta en optimas condiciones';
      else
        fprintf('Mecánico \n');
        estado='Mecánico';
      endif
      %Conectar con PGADMIN
      conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
      'port','5432','user','postgres','password','jblue5queso'));
      params=cell(1,3);
      params{1,1}=modelo;
      params{1,2}=recorrido;
      params{1,3}=estado;
      N=pq_exec_params(conn, "insert into taxi(modelo, recorrido_km, estado) values($1,$2,$3);",params); %insertar datos en la tabla
      %Texto
      save('taxi.txt','modelo', 'recorrido','estado','-ascii');
    endif
    if i==2
       conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
       'port','5432','user','postgres','password','jblue5queso'));
       N=pq_exec_params(conn, "Select * from taxi;");%historial de la tabla
       disp(N);
       retry=false;
    endif
   catch
      fprintf("Solo se permiten valores numericos, vuelava a selecionar %d \n");
      msg = lasterror.message;
      fprintf(msg);
    end_try_catch
 endwhile